Lib/fontTools/ttLib/tables/_k_e_r_n.py

```diff
@@ -85,26 +87,41 @@ def fromXML(self, name, attrs, content, ttFont):
 		if format not in kern_classes:
 			subtable = KernTable_format_unkown(format)
 		else:
-			subtable = kern_classes[format]()
+			apple = self.version == 1.0
+			subtable = kern_classes[format](apple)
 		self.kernTables.append(subtable)
 		subtable.fromXML(name, attrs, content, ttFont)
 
 
 class KernTable_format_0(object):
 
+	# 'version' is kept for backward compatibility
+	version = format = 0
+
+	def __init__(self, apple=False):
+		self.apple = apple
+
 	def decompile(self, data, ttFont):
-		version, length, coverage = (0,0,0)
 		if not self.apple:
 			version, length, coverage = struct.unpack(">HHH", data[:6])
+			if version != 0:
+				from fontTools.ttLib import TTLibError
+				raise TTLibError(
+					"unsupported kern subtable version: %d" % version)
+			tupleIndex = None
+			# Should we also assert length == len(data)?
 			data = data[6:]
 		else:
-			version, length, coverage = struct.unpack(">LHH", data[:8])
+			length, coverage, tupleIndex = struct.unpack(">LHH", data[:8])
```

Comment:
we were also decompiling `coverage` incorrectly here for apple kern subtables: the `tupleIndex` was being read in as "coverage" attribute, whereas the actual `coverage` (in second position, where `length` used to be) was being thrown away.